eslint-plugin-local-rules/rules/no-direct-validate-calls.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-env node */
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    fixable: 'code',
+    docs: {
+      description: 'Require _requestValidation() instead of direct validate() calls to respect manual validation mode',
+    },
+    schema: [],
+  },
+  create(context) {
+    return {
+      MemberExpression(node) {
+        if (
+          node.object.type === 'ThisExpression' &&
```

Comment:
Should we also cover cases like this:
```
this.getSomeChildElement().validate();
```

Reply:
Regardless of the practical usages today, is there any reason to prevent all `validate` calls on anything?

Reply:
That could work too! I remember considering this, but I can't recall why I decided against this. There are plenty of usages of validate() in tests, but we can just turn off the rule in the test folder.

Reply:
BTW, I'm currently checking if we could implement this restriction with an existing ESLint plugin like no-restricted-properties instead of building our own.

Reply:
Seems to be possible with no-restricted-syntax:

```json
"no-restricted-syntax": [
  "error",
  {
    "selector": "CallExpression[callee.property.name='validate']",
    "message": "Don't call validate() directly - it bypasses manual validation mode. Use _requestValidation() instead"
  }
]
```

Reply:
Replaced the custom implementation with `no-restricted-syntax`. The downside is that this rule can only be disabled as a whole. I also noticed that eslint-config-vaadin has a [default configuration](https://github.com/vaadin/eslint-config-vaadin/blob/edde9d90defdd001405210733dd66339b8e8b1c1/src/rules/eslint/suggestions.ts#L129-L145) for this rule, so I copied the parts that I feel are relevant for our repository.

Reply:
Thanks, seems to work well.